### PR TITLE
#307 adds tests to the blobs module

### PR DESCRIPTION
**Pull request by betarelease**
## PR Checklist

- [x ] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
- [x] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
- [x] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
- [x] I've not introduced any new known security vulnerabilities. I've run `cargo audit`

- [x] I've included a brief description and a good title of the proposed changes and how to test them.
- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
- [x] I've checked that the code is up-to-date with the `pyrsia/main` branch.
- [x] I've read ["What is a Good PR?"](good_pr.md)
- [x] I've assigned this Pull Request to "pyrsia/collaborators"

## Description

Fixes pyrsia/pyrsia#307

This PR adds tests to the blobs module and refactors the concern for handling the storage of the blob into its own file

### CODE REVIEW

**Comment by betarelease** [comment]
@MitaliBo Can you take a look at the changes I made here? I want to get this merged and follow this up with a PR so that we can optimize the download and save of the docker blobs as a different issue.

File: src/docker/v2/models/blobs.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
+/*
```
Comment (MitaliBo) [inline src/docker/v2/models/blobs.rs:1]: why was blobs code moved inside models? this is not relevant module

File: readme.md [hunk 2]
```diff
@@ -92,11 +92,35 @@ For now Pyrsia only supports Docker artifacts. Follow these steps to run a Pyrsi
...
+    cargo run --bin pyrsia_node -- -p 7888
+
+    # RUST_LOG=debug cargo run --bin pyrsia_node -- -p 7888 # Use this environment variable if you would like to see debug logs
+    ```
+
+    ```
+    cargo run --bin pyrsia_node -- -p 8181
+
+    # RUST_LOG=debug cargo run --bin pyrsia_node -- -p 8181 # Use this environment variable if you would like to see debug logs
+    ```
```
Comment (MitaliBo) [inline readme.md:111]: this file does not look updated

**Comment by MitaliBo** [comment]
@betarelease i think we need to do some cleanup in this PR and this not well refactored as its just duplicated the code in another module which is actually disconnected with rest of the code. It actually does not fulfill the requirement of #307 so we should remove this with that issue otherwise that issue will get closed if this gets merged.

File: readme.md (hunk 2 above)
Comment (betarelease) [reply readme.md:111]: I will update this. Last time I updated was > 1 week ago

File: src/docker/v2/models/blobs.rs (hunk 1 above)
Comment (betarelease) [reply src/docker/v2/models/blobs.rs:1]: I think I missed removing the original code from handlers. But the idea is handlers should only do the handling of the API and model goes and does the saving.(like Model and Controller - handler is the controller in this case).

Also this helps us separate the concern of communicating with dockerhub(which is what handlers does) and storing(which is what the model does)

This provides cleaner interface for testing and independent testing. We can then swap the storage model with whatever else we decide to use(say s3 for example) or a different way of storing blobs without touching handlers.

**Comment by betarelease** [comment]
closing this since I opened #431

**Pull request closed without merging**